Automattic/wp-calypso
# Extract webpack and babel tools from ./client/ into its own directory

#### Background

Right now `./client` has code that runs bundled on the browser, code that runs on the server and code used to build Calypso. 

This mixture of different stages (build vs runtime) and in different environments (browser vs node) creates some confusion and complexity. For example, it is hard to come up with a set of eslint rules without having to apply exceptions everywhere (some examples are `import/no-nodejs-modules` or `no-console`) or a good definition of what a `devDependency` is (for example, one could argue that `webpack-dev-server` is a `devDependency` for server code, would be a `dependency` for build-tools code and nothing for browser code)

#### Changes proposed in this Pull Request

This PR extract Babel and Webpack tools that we have in `./client/webpack-tools` and `./client/server/bundler` into its own `./build-tools` directory.

#### Feedback wanted

Is `./build-directory` the best place for those tools? 

* What about `packages/calypso-build`? The seems mostly related to support "modern versions of JavaScript (ESNext and JSX) and SASS", while these tools are more generic.
* What about a different package? Some tools like `assets-writer` can have some uses outside Calypso, but other tools (like `section-loader`) are very coupled to Calypso's build process, so it wouldn't make a lot of sense to make a public package. Maybe a private package?
* What about multiple packages? That could work too.
* Something else?


#### Testing instructions

* Verify calypso.live still works
* Verify all tests are green
